crates/matrix-sdk-ui/src/timeline/controller/state.rs

```diff
@@ -720,6 +738,18 @@ impl TimelineStateTransaction<'_> {
     fn adjust_day_dividers(&mut self, mut adjuster: DayDividerAdjuster) {
         adjuster.run(&mut self.items, &mut self.meta);
     }
+
+    fn replace_all_events_encryption(&mut self, is_encrypted: bool) {
```

Comment:
That's probably a really poor name for this method, at least I can't really explain from the name what this method does.

Nor is it quite clear that this method does enough, it's just setting a flag on the event that the room is encrypted? How will this trigger shield replacements?

Reply:
I can add some docs for it and maybe find a new name, but it's kind of difficult, since all it does is replace the `is_room_encrypted` value for each timeline item and create a `VectorDiff::Set` for each item so this update is emitted.